lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -877,4 +892,79 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     */
+    private function getOrderByForOver(string $query) : string
+    {
+        //determine if 'ORDER BY' is part of the query
+        $orderByPosition = strripos($query, 'order by');
+
+        // early return if ORDER BY not found in query string
+        if ($orderByPosition === false) {
+            return '';
+        }
+
+        // build dictionary if available
+
+        $selectedColumns = preg_split('/[, ]/', substr($query, 0, $orderByPosition -1));
+
+        // filter out 'AS'
+        $filteredSelectedColumns = array_filter(
+            // split selected columns
+            $selectedColumns,
+            function ($element) {
+                // don't return 'AS' and empty elements
+                return strtoupper($element) !== 'AS' && trim($element !== '') && $element !== false;
```

Comment:
That's what the "testing" tool said... aka. phpcs/stan/whatever is used

Reply:
If UNION might be a problem, I will develop against it and provide a fix for it.
Either way... Doctrine by itself is USELESS on an IBM system as it is, I am not talking about DB2 by itself. I am talking about about IBMi.

Reply:
>That's what the "testing" tool said... aka. phpcs/stan/whatever is used

`preg_split` can return `false` as a return value, for instance, if a regex is invalid. But an element itself cannot be `false` as far as I can tell.

Reply:
> If UNION might be a problem, I will develop against it and provide a fix for it.

The `UNION` itself is not necessarily a problem since DBAL doesn't support it very well. I just made it up to illustrate where a potential problem may be. But other queries with sub-queries and multiple `ORDER BY`'s may be a real problem.

> Either way... Doctrine by itself is USELESS on an IBM system as it is, I am not talking about DB2 by itself. I am talking about IBMi.

Most likely, nobody tested DBAL on IBMi or at least it's not being tested continuously — that's for sure. If we decide to support it, is there a way to set it up in a more or less convenient way? For instance, I use [ibmcom/db2express-c](https://hub.docker.com/r/ibmcom/db2express-c/) for testing on Linux.